Use relative path for offense message in `Lint/DuplicateMethods`

pocke opened the pull request:
Currently `Lint/DuplicateMethods` cop messages have absolute paths.

```bash
$ rubocop test.rb --only Lint/DuplicateMethods
Inspecting 1 file
W

Offenses:

test.rb:3:1: W: Method Object#foo is defined at both /home/pocke/ghq/github.com/bbatsov/rubocop/test.rb:1 and /home/pocke/ghq/github.com/bbatsov/rubocop/test.rb:3.
def foo
^^^

1 file inspected, 1 offense detected
```

I think Absolute path is unnecessarily long.
And we use relative path for a location of an offense(e.g. `test.rb:3:1`).
So, I think we should use relative path for this message also.

By this change, the absolute path is replaced with relative path.

For example:

```bash
$ rubocop test.rb --only Lint/DuplicateMethods
Inspecting 1 file
W

Offenses:

test.rb:3:1: W: Method Object#foo is defined at both test.rb:1 and test.rb:3.
def foo
^^^

1 file inspected, 1 offense detected
```

What do you think?


-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

rrosenblum commented:
Makes sense to me.

bbatsov commented:
Ideally, yes. And we should check whether other cops aren't using absolute paths for consistency's sake.

pocke commented:
I've added tests, and changed to use `smart_path` method. The method changes path to relative only if file is in project root. 



> Should there be a test that shows off this functionality?

Thanks for your suggestion. I've added tests.

> And we should check whether other cops aren't using absolute paths for consistency's sake.


```
$ git grep buffer.name
lib/rubocop/comment_config.rb:      Cop::Cop.qualified_cop_name(cop_name.strip, processed_source.buffer.name)
lib/rubocop/cop/commissioner.rb:        remove_irrelevant_cops(processed_source.buffer.name)
lib/rubocop/cop/layout/leading_comment_space.rb:                    config_ru?(processed_source.buffer.name)
lib/rubocop/cop/lint/duplicate_methods.rb:          path = smart_path(range.source_buffer.name)
lib/rubocop/cop/style/file_name.rb:          file_path = processed_source.buffer.name
lib/rubocop/cop/team.rb:          filename = buffer.name
lib/rubocop/processed_source.rb:      buffer_name = @path || STRING_SOURCE_NAME
lib/rubocop/processed_source.rb:      @buffer = Parser::Source::Buffer.new(buffer_name, 1)
```

`Style/FileName` cop uses `buffer.name`, but the cop uses relative paths for messages.
So, I guess cops that use absolute paths for messages doesn't exist.

bbatsov commented:
👍
